packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/Converters/FrameworkToJsiiConverter.cs

```diff
@@ -251,6 +251,10 @@ protected override bool TryConvertMap(IReferenceMap referenceMap, TypeReference
                     return false;
                 }
 
+                if (convertedElement != null && !(convertedElement is String) && !convertedElement.GetType().IsPrimitive)
```

Comment:
_ This can be cleaner, I feel. But doing so requires for extensive refactoring, so whatever.

Reply:
Yup. This felt slightly dirty, but indeed, I couldn't afford that kind of time right now.